halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/GenerateCommand.java

```diff
@@ -1,4 +1,37 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.netflix.spinnaker.halyard.cli.command.v1;
 
-public class GenerateCommand {
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.DaemonService;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+public class GenerateCommand extends NestableCommand {
+  @Getter(AccessLevel.PUBLIC)
+  private String commandName = "generate";
+
+  @Getter(AccessLevel.PUBLIC)
+  private String description = "Generate the full Spinnaker config for you current deployment.";
+
+  @Override
+  protected void executeThis() {
+    DaemonService service = Daemon.getService();
+    String currentDeployment = service.getCurrentDeployment();
+    service.generateDeployment(currentDeployment, "", false);
```

Comment:
"where the trying" should heave read "where the tying".

What I'm getting at is since config wont typically change substantially between versions I wouldnt expect the config and version to be intrinsically coupled together. Especially since the most likely config change from one version to the next would be incremental addition of config for a feature added to the new version.

I imagine one would have multiple configs because one has (or is preparing for) multiple deployments or is holding on to something older to revert back to either because it is a change in the conceptual configuration or from a version update.

Intuitively I'd expect configurations to have an intent that is version independent (e.g. production vs testing where you're going to use the same fundamental policies and approach as you evolve from version to version). So I'd expect configurations to have names that reflect that (as you do). Specific versions might require specific deviations from the policy because of new features added or old features deprecated. Given that I'd expect you to have a set of policies/intent configurations that I'm calling baseline configurations, then a set of versions you want to support which might have "configuration deltas" that override (or add to) the baseline. These would probably be per policy.

Maybe it doesnt matter because you never look at the raw config, the tooling provides the diffs, and when you create a new version, you start with the last version and modify it. So you get to the same place from a UX standpoint. It might be more work and complex this way, but you only have one type of config so the processing of the config itself is probably simpler.

Just thinking out loud.